### DISCUSSION TITLE
Nesting multiple repos under one repo

### DISCUSSION THREAD

**Discussion by RainyCityCoder** [community/community#33747] [Other Feature Feedback, Questions, & Ideas]
For the past several days I've been trying to consolidate my class's assignment repos by nesting individually under a repo of the class name. Most of the resources I've found so far, or are suggested to me, are designed to merge two peer repos into one. This isn't what I want. My goal is to nest the assignment repos under a class repo both locally and remotely (and as I understand it; submodules would still be remotely distinct repos, which I don't want):

Now:
1. assignment 1 repo
2. assignment 2 repo
3. assignment 3 repo

Desired:
1. class repo
    - assignment 1 code, branches, commits
    - assignment 2 code, branches, commits
    - assignment 3 code, branches, commits

I attempted to use [this](https://medium.com/@checko/merging-two-git-repositories-into-one-preserving-the-git-history-4e20d3fafa4e) article's instructions, repeating step 4 for assignment 2 & 3's repos, but only the `master` branch was pulled down, and when I run `~/class-repo/assignment-#-repo$ git pull`; an error is returned: `error: Pulling is not possible because you have unmerged files.`. 

`~/class-repo$ git merge --continue` doesn't fix this, it returns: `error: Committing is not possible because you have unmerged files`. 

`~/class-repo$ git fetch --all` hasn't pulled down all branches for each assignment repo. 

How do I preserve the commits and branches of each assignment repo while nesting them in this fashion? I keep hitting brick walls, so if it's not possible to do, I understand.

**Comment by airtower-luna** [comment]
> How do I preserve the commits and branches of each assignment repo while nesting them in this fashion? I keep hitting brick walls, so if it's not possible to do, I understand.

It's not possible. If you want to keep branches, commits, etc. independent of each other you need separate repositories. You cannot have a repository in a repository. Well, with the exception of submodules, but those do that only in the working tree, not the repository data.

The article you linked would result in one repository with all the histories merged together.

**Reply by RainyCityCoder** [reply]
I feared so. Thank you.

**Comment by jfullstackdev** [comment]
would forks not be sufficient ? they can have their own commits and you can simply compare that to the `main` branch which is I think the template of the project, so you can easily see the tasks done by the students and you should not merge every change, so you can separate each
